src/app/recording/Recording.js

```diff
@@ -331,6 +323,16 @@ function Recording () {
     setSelectedSegmentId(createdFeature.id)
   }
 
+  async function onBoundsChange(bounds) {
+    const topRight = `${bounds._northEast.lng},${bounds._northEast.lat}`
+    const bottomRight = `${bounds._southWest.lng},${bounds._northEast.lat}`
+    const bottomLeft = `${bounds._southWest.lng},${bounds._southWest.lat}`
+    const topLeft = `${bounds._northEast.lng},${bounds._southWest.lat}`
+    const boundingBox = `${topRight},${bottomRight},${bottomLeft},${topLeft},${topRight}`
+    const geojson = await getSegments(boundingBox)
+    setGeoJson(geojson)
+  }
```

Comment:
This is pretty ugly :(

Reply:
Yes, we shouldn't discard all previously loaded segments. The least thing we could do here is a merge of the prev and next segments.

Also, we should only request segments if the changed bbox is outside of the bboxes we already requested, right?

Reply:
Oh btw: The map (PTMap) itself filters the segments again and only keeps those that are in the bounding box. This means, we really can put in here as many segments as we want to.

Reply:
@henrymoews OK sure, I think at some point it also becomes an issue to request so many segments from the backend, right now it's fine with a small number but eventually it will be very inefficient to request the whole collection.